install.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+# This is an install script for notes
+
+function assertInstalled() {
+    for var in "$@"; do
+        if ! which $var &> /dev/null; then
+            echo "$var is required but not installed, exiting."
+            exit 1
+        fi
+    done
+}
+
+# Yay, Echo self documents! :D
+echo "Checking for root..."
+    [ "$(whoami)" != "root" ] && exec sudo -- "$0" "$@"
+
+# Make sure we have everything actually installed that we need to install this.
+echo "Checking for Dependencies..."
+    assertInstalled bash curl tar mktemp install make
+
+# Variable Definitions go here. 
+user_home=`eval echo ~$SUDO_USER`
+bash_completion_dir=`pkg-config --variable=completionsdir bash-completion 2>/dev/null`
```

Comment:
Well, we check bash_completions here, then use it in the makefile. I could most likely move this into Makefile